types/priv_validator/socket_tcp.go

```diff
@@ -0,0 +1,66 @@
+package types
+
+import (
+	"net"
+	"time"
+)
+
+// timeoutError can be used to check if an error returned from the netp package
+// was due to a timeout.
+type timeoutError interface {
+	Timeout() bool
+}
+
+// tcpTimeoutListener wraps a *net.TCPListener to standardise protocol timeouts
+// and potentially other tuning parameters.
+// Implements net.Listener.
```

Comment:
you can check that in compile time `var _ net.Listener = (*tcpTimeoutListener)(nil)`